native/lib.rs

```diff
@@ -22,11 +22,12 @@ use std::{
 #[rustler::nif(name = "create")]
 fn create_3(env: Env, cache_name: Atom, max_size: usize, min_q1_size: usize) -> Atom {
     // let _timer = DropTimer::new();
-    let mut caches = CACHES.lock().unwrap();
     let cache_name = HBin::from_atom(env, cache_name);
-    if caches.get(&cache_name).is_some() {
+    let caches = CACHES.read().unwrap();
```

Comment:
I think grabbing the read lock, dropping the lock, and then acquiring the write lock introduces a race condition where two threads can try to insert the same cache at the same time. There's probably not real consequence to that for create. Maybe use an atomic read -> write upgradable lock or start with a write lock.